src/bokeh/models/glyphs.py

```diff
@@ -859,6 +860,27 @@ def __init__(self, *args, **kwargs) -> None:
         The color mapping step happens on the client.
     """)
 
+    def _validate_with_data_source(self, source: ColumnDataSource):
+        class_name = self.__class__.__name__
+
+        if not isinstance(self.color_mapper, WeightedStackColorMapper):
+            raise ValueError(f"{class_name}.color_mapper should be a WeightedStackColorMapper not a {self.color_mapper.__class__.__name__}")
+        n_colors = len(self.color_mapper.palette)
+        if n_colors < 1:
+            # This check should really be in ColorMapper.
+            raise ValueError(f"{class_name}.color_mapper.palette does not contain any colors")
+
+        image = source.data[self.image] if isinstance(self.image, str) else self.image
```

Comment:
I was about to write that this is incorrect, because:
```py

In [1]: from bokeh.models import *

In [2]: i = ImageStack()

In [3]: i.image
Out[3]: Field(field='image', transform=Unspecified, units=Unspecified)
```
however
```py
In [4]: i.image = "foo"
i
In [5]: i.image
Out[5]: 'foo'
```
I would have expected this to be normalized to `Field`.

Reply:
Similar issue with the value case. Note that this is a dataspec, so it should be also an expression.

Reply:
Note also that the `image` property definition is completely wrong:
```py
image = NumberSpec(default=field("image"), help="""
```
which doesn't actually allow ndarrays as items, but numbers. I'm pretty sure we don't have a dataspec type that would support ndarrays. I would probably start with that, because currently we can't assign a scalar value to `image`, nor validate that case. Having that, the validator could be reused in the contextual validator of the glyph.

Reply:
> I'm pretty sure we don't have a dataspec type that would support ndarrays.

I never got around to implementing one, so I suspect this is correct. A proper dataspec for these more complex column types would be welcome, and I think probably a little more straightforward to implement sanely nowadays than it would have been in the early days.